benchmark/legateboost_scaling.py

```diff
@@ -31,6 +31,9 @@ def create_dataset(args):
     gen = cn.random.Generator(cn.random.XORWOW(seed=42))
     rows = args.nrows if args.strong_scaling else args.nrows * n_processors
     X = gen.normal(size=(rows, args.ncols), dtype=cn.float32)
+    X[:, args.ncols // 2 :] = gen.binomial(1, 0.5, size=(rows, args.ncols // 2)).astype(
+        cn.float32
+    )
```

Comment:
Yes the benchmark in this form favors the shared memory version - although this PR is faster even with all numerical features. I did this because recent experiments on the mortgage dataset shows a 6x performance hit for the global atomics version. Its really important that we reflect some low cardinality features in our benchmarking as I think very few datasets are all numerical. 50% is not an unreasonable distribution based on many datasets we see. I will set a parameter for the ratio of numerical to binary features and default it to 50%.